PHIOS-6358 [Motion Free] Q3 Implement Platform virtual vehicle API support for Onboarding

alex-kepar opened the pull request:
- added supporting values "MojioType" and "VehicleType" according to https://1mojio.atlassian.net/wiki/spaces/AR/pages/1200259114/Motion-Free+APIs#POST-{domain-prefix}-api.moj.io/mojios/{mojioType}

https://1mojio.atlassian.net/browse/PHIOS-6358

alex-kepar committed:
PHIOS-6358 [Motion Free] Q3 Implement Platform virtual vehicle API support for Onboarding:
- added supporting values "MojioType" and "VehicleType" according to https://1mojio.atlassian.net/wiki/spaces/AR/pages/1200259114/Motion-Free+APIs#POST-{domain-prefix}-api.moj.io/mojios/{mojioType}

alex-kepar committed:
PHIOS-6358 [Motion Free] Q3 Implement Platform virtual vehicle API support for Onboarding:
- added supporting values "MojioType" and "VehicleType" according to https://1mojio.atlassian.net/wiki/spaces/AR/pages/1200259114/Motion-Free+APIs#POST-{domain-prefix}-api.moj.io/mojios/{mojioType}

alex-kepar committed:
PHIOS-6358 [Motion Free] Q3 Implement Platform virtual vehicle API support for Onboarding:
- added supporting values "MojioType" and "VehicleType" according to https://1mojio.atlassian.net/wiki/spaces/AR/pages/1200259114/Motion-Free+APIs#POST-{domain-prefix}-api.moj.io/mojios/{mojioType}

andriimakukha commented on MojioSDK/Models/Vehicles/Vehicle.swift:
```diff
@@ -16,6 +16,18 @@
+    case unknown = "Unknown"
+
+    public init(from decoder: Decoder) throws {
+        let label = try decoder.singleValueContainer().decode(String.self)
```
Maybe `value` or `stringValue` is better?

alex-kepar replied:
@andriimakukha  Thanks. It renamed to rawValue.

andriimakukha commented on MojioSDK/Models/Vehicles/Mojio.swift:
```diff
@@ -16,6 +16,17 @@
+    case unknown = "Unknown"
+
+    public init(from decoder: Decoder) throws {
+        let label = try decoder.singleValueContainer().decode(String.self)
```
Maybe `value` or `stringValue` is better?

alex-kepar replied:
@andriimakukha  Thanks. It renamed to rawValue.

alex-kepar committed:
PHIOS-6358 [Motion Free] Q3 Implement Platform virtual vehicle API support for Onboarding:
- added supporting values "MojioType" and "VehicleType" according to https://1mojio.atlassian.net/wiki/spaces/AR/pages/1200259114/Motion-Free+APIs#POST-{domain-prefix}-api.moj.io/mojios/{mojioType}
